src/main/java/org/opensearch/knn/indices/ModelDao.java

```diff
@@ -362,11 +363,13 @@ public Model get(String modelId) throws ExecutionException, InterruptedException
             /*
                 GET /<model_index>/<modelId>?_local
             */
-            GetRequestBuilder getRequestBuilder = new GetRequestBuilder(client, GetAction.INSTANCE, MODEL_INDEX_NAME).setId(modelId)
-                .setPreference("_local");
-            GetResponse getResponse = getRequestBuilder.execute().get();
-            Map<String, Object> responseMap = getResponse.getSourceAsMap();
-            return Model.getModelFromSourceMap(responseMap);
+            try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
```

Comment:
Can you please add some comment which can explain what we want to achieve here? I also don't see the usage of context variable. May be I am missing something here.